infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -708,4 +728,47 @@ public void shutdown() {
     private String getLocalHost() {
         return NodeLocator.parseString(serverContext.getLocalEndpoint()).getHost();
     }
+
+    private LockClient createLockClient(LockStateType initState) throws Exception {
```

Comment:
May define a LockConfig or LockUtility class and move this function there.

Reply:
I thought about it too but it made moving it difficult because some of the things to create a lock are located in this class - for example, Discovery service is being passed here, this will create circular dependencies. I think since this is for testing, its fine to have it there for now. Otherwise, it will entail longer refactoring.